javascript/react/good-to-know/immutability-helpers-in-react.md

```diff
@@ -25,52 +25,73 @@ aspects:
 
 ---
 
-# Immutability helpers in **React**
+# Immutability helpers in React
 
 ---
 ## Content
 
-Opting out for an immutable style of data management, a significant speed up can be achieved by easily implementing a fast `shouldComponentUpdate()` lifecycle method.
+When opting in for an immutable style of data management, a significant speedup can be achieved by implementing a `shouldComponentUpdate()` lifecycle method. To do this, you would need a method that aids with comparing *versions* of your component's `state`.
 
-Given the fact that **JavaScript** wasn't designed for dealing with immutable data, **React** provides an immutability helper `update()` with a  syntax inspired by **MongoDB**.
-
-Mutate without having any way of determining which data has changed:
+Let's take a look at the following example of mutating data:
 
 ```javascript
-myData.x.y.z = 7;
-myData.a.b.push(9);
+myData.x.y.z = 10;
+// or...
+myData.a.b.push(13);
+```
+
+Given the fact that **JavaScript** wasn't designed for dealing with immutable data, using one of the above methods does not provide a way of determining which data has changed since the previous copy. To help with this, **React** used to provide an immutability helper (has since become legacy) which is now replaced by the `immutability-helper` library.
```

Comment:
```suggestion
Given the fact that **JavaScript** wasn't designed for dealing with immutable data, using one of the above methods does not provide a way of determining which data has changed after the mutation. To help with this, **React** used to provide an immutability helper (has since become legacy) which is now replaced by the `immutability-helper` library.
```